pkg/api/v1/types.go

```diff
@@ -72,6 +72,8 @@ type ObjectMeta struct {
 	// definition.
 	// Cannot be updated.
 	// More info: http://releases.k8s.io/HEAD/docs/user-guide/identifiers.md#names
+	// +k8s:openapi-gen=validate(Len(5,100))
```

Comment:
I'd argue that we should have a library of validators that are more expressive.  When it comes time to do this for real, it would be cleaner to say "LessThan(10)" and "Between(5, 100)" and "GreaterThanOrEqualTo(0)" to produce the best error messages.  Obviously those functions need to be type-neutral, so internally probably have to call at least signed and unsigned versions on int64 and uint64